### DISCUSSION TITLE
`CapsuleReader` async usage

### DISCUSSION THREAD

**Discussion by busslina** [GregoryConrad/rearch-dart#201] [Q&A] [answered]
Is it secure to use a `CapsuleReader` asynchronously like we do with `BuildContext` on a button pressed callback?

**Comment by GregoryConrad** [comment, marked as answer]
No, that should be giving you a warning in the console.

You should move the `use(capsule)` to the top of the build method and then reference it in the callback instead.

**Reply by busslina** [reply]
Thanks
